org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/IndexServiceClientTest.xtend

```diff
@@ -225,19 +270,69 @@ class IndexServiceClientTest {
 			"dropping type-incompatible element (expected eReference type: Grammar; index service provided element of type: Condition).")
 	}
 
+	@Test
+	def void shouldWarnAboutMissingContextRequest() {
+		// given
+		val expected = mock(IEObjectDescription)
+		when(expected.EClass).thenReturn(XtextPackage.eINSTANCE.grammar)
+
+		val unitUnderTest = mockedIndexService("Sample content", #[expected], null)
+		val resource = mockedResource("Sample content", new BasicEList(#[mock(EObject)]))
+
+		val reference = XtextPackage.eINSTANCE.grammar_UsedGrammars
+
+		// when
+		unitUnderTest.getScope(resource, reference, null)
+
+		// then
+		verify(logAppender).doAppend(logCaptor.capture)
+		assertThat(logCaptor.value.formattedMessage).isEqualTo(
+			"Failed to retrieve context request. Request to index service will be sent without authorization header.")
+	}
+
+	@Test
+	def void shouldWarnAboutMissingAuthenticationHeader() {
+		// given
+		val expected = mock(IEObjectDescription)
+		when(expected.EClass).thenReturn(XtextPackage.eINSTANCE.grammar)
+
+		val unitUnderTest = mockedIndexService("Sample content", #[expected], mock(HttpServletRequest))
+		val resource = mockedResource("Sample content", new BasicEList(#[mock(EObject)]))
+
+		val reference = XtextPackage.eINSTANCE.grammar_UsedGrammars
+
+		// when
+		unitUnderTest.getScope(resource, reference, null)
+
+		// then
+		verify(logAppender).doAppend(logCaptor.capture)
+		assertThat(logCaptor.value.formattedMessage).isEqualTo(
+			"Context request carries no authorization header. Request to index service will be sent without authorization header.")
+	}
+
 	private def mockedIndexService(String payload, List<IEObjectDescription> expected) {
+		val request = mock(HttpServletRequest)
+		when(request.getHeader(eq(AUTHORIZATION))).thenReturn(AUTH_HEADER)
+
+		return this.mockedIndexService(payload, expected, request)
+	}
+
+	private def mockedIndexService(String payload, List<IEObjectDescription> expected,
```

Comment:
expected is misleading here, since this is not expected but actually returned, right?

Reply:
Yeah, renamed accordingly